sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -337,6 +337,27 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     checkAnswer(lowerCaseData.intersect(upperCaseData), Nil)
   }
 
+  test("intersect - nullability") {
```

Comment:
I think it's probably OK since we can also `checkAnswer` here to verify the query result.